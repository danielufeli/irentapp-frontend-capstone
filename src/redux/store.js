import { configureStore } from '@reduxjs/toolkit';
import housesReducer from '../features/houses/housesState';

const store = configureStore({
  reducer: {
    houseList: housesReducer,
  },
});

export default store;
