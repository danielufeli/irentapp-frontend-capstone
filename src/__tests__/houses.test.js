import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Houses from '../components/houses/houses';
import store from '../redux/store';

describe('App', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <BrowserRouter>
          <Houses />
        </BrowserRouter>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
