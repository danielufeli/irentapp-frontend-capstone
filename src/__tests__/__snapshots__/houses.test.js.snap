// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container d-flex flex-column flex-md-row"
      >
        <aside
          class="navbar navbar-expand-md navbar-light d-flex flex-md-column"
        >
          <a
            href="https://www.google.com/"
          >
            <img
              alt="logo"
              height="200"
              src="logo.png"
              width="200"
            />
          </a>
          <button
            aria-controls="navbarSupportedContent"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-bs-target="#navbarSupportedContent"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse w-100"
            id="navbarSupportedContent"
          >
            <ul
              class="navbar-nav w-100 d-flex flex-md-column text-center text-md-end"
            >
              <li>
                <a
                  href="/houses"
                >
                  Houses
                </a>
              </li>
              <li>
                <a
                  href="/addHouse"
                >
                  Add House
                </a>
              </li>
              <li>
                <a
                  href="/reservations"
                >
                  My Reservations
                </a>
              </li>
              <li>
                <a
                  href="/add_reservation"
                >
                  Add Reservation
                </a>
              </li>
              <li>
                <button
                  type="submit"
                >
                  Log Out
                </button>
              </li>
            </ul>
          </div>
        </aside>
        <main
          class="ps-0 ps-md-5 flex-grow-1 pt-3"
        >
          <div
            class="container cards"
          >
            <div
              class="react-multi-carousel-list carousel-container carousel"
              dir="ltr"
            >
              <ul
                class="react-multi-carousel-track "
                style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
              />
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container d-flex flex-column flex-md-row"
    >
      <aside
        class="navbar navbar-expand-md navbar-light d-flex flex-md-column"
      >
        <a
          href="https://www.google.com/"
        >
          <img
            alt="logo"
            height="200"
            src="logo.png"
            width="200"
          />
        </a>
        <button
          aria-controls="navbarSupportedContent"
          aria-expanded="false"
          aria-label="Toggle navigation"
          class="navbar-toggler"
          data-bs-target="#navbarSupportedContent"
          data-bs-toggle="collapse"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="collapse navbar-collapse w-100"
          id="navbarSupportedContent"
        >
          <ul
            class="navbar-nav w-100 d-flex flex-md-column text-center text-md-end"
          >
            <li>
              <a
                href="/houses"
              >
                Houses
              </a>
            </li>
            <li>
              <a
                href="/addHouse"
              >
                Add House
              </a>
            </li>
            <li>
              <a
                href="/reservations"
              >
                My Reservations
              </a>
            </li>
            <li>
              <a
                href="/add_reservation"
              >
                Add Reservation
              </a>
            </li>
            <li>
              <button
                type="submit"
              >
                Log Out
              </button>
            </li>
          </ul>
        </div>
      </aside>
      <main
        class="ps-0 ps-md-5 flex-grow-1 pt-3"
      >
        <div
          class="container cards"
        >
          <div
            class="react-multi-carousel-list carousel-container carousel"
            dir="ltr"
          >
            <ul
              class="react-multi-carousel-track "
              style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
            />
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
